src/Clock.tsx

```diff
@@ -0,0 +1,61 @@
+/* eslint-disable no-console */
+import React from 'react';
+
+type State = {
+  date: Date,
+};
+
+type Props = {
+  clockName: string,
+};
+
+export class Clock extends React.Component<Props, State> {
+  state = {
+    date: new Date(),
+  };
+
+  timerId = 0;
+
+  componentDidMount() {
+    this.timerId = window.setInterval(() => {
+      const today = new Date();
+
+      console.info(today.toUTCString().slice(-12, -4));
+      this.setState({ date: today });
+    }, 1000);
+  }
+
+  componentDidUpdate(prevProps: Readonly<Props>): void {
+    const { clockName } = this.props;
+
+    if (clockName !== prevProps.clockName) {
+      console.debug(`Renamed from ${prevProps.clockName} to ${clockName}`);
+    }
+  }
```

Comment:
Could go for `prevProps` also as you did with current
```suggestion
  componentDidUpdate(prevProps: Readonly<Props>): void {
    const { clockName: prevClockName } = prevProps;
    const { clockName } = this.props;

    if (clockName !== prevClockName) {
      console.debug(`Renamed from ${prevClockName} to ${clockName}`);
    }
  }
```